src/main/java/no/unit/nva/cristin/projects/CristinApiClient.java

```diff
@@ -75,21 +67,24 @@ protected URL generateQueryProjectsUrl(Map<String, String> parameters) throws Ma
 
     protected URL generateGetProjectUrl(String id, String language) throws MalformedURLException, URISyntaxException {
         URI uri = new URIBuilder()
-                .setScheme(HTTPS)
-                .setHost(CRISTIN_API_HOST)
-                .setPath(CRISTIN_API_PROJECTS_PATH + id)
-                .addParameter("lang", language)
-                .build();
+            .setScheme(HTTPS)
+            .setHost(CRISTIN_API_HOST)
+            .setPath(CRISTIN_API_PROJECTS_PATH + id)
+            .addParameter("lang", language)
+            .build();
         return uri.toURL();
     }
 
-    protected static <T> T fromJson(InputStreamReader reader, Class<T> classOfT) throws IOException {
-        try {
-            return new Gson().fromJson(reader, classOfT);
-        } catch (JsonSyntaxException e) {
-            final String s = e.getMessage() + " " + reader;
-            throw new IOException(s, e);
-        }
+    private List<Project> enrichProjects(String language, List<Project> projects) {
+        return projects.stream()
+            .map(project -> {
```

Comment:
In general, I react to the code in this class (it isn't code you have written) for a number of reasons:

 - it should take account for the fact that we want to download the projects from the Cristin API. rapidly, probably in parallel, but the code will lock for each request, accumulating time for each request
 - we have very little control over the requests because we are relying on URL.openStream
 - failures are ignored, which is unfortunate
 - we should have a chat with Orestis about how to avoid these issues practically in Lambdas (I have a code example, based on CompletableFuture, which I think will work, BUT may have side effects in lambdas)

Reply:
Maybe create a new Jira issue where we will improve it?

Reply:
Opened a issue on this